kobo/worker/main.py

```diff
@@ -81,6 +90,10 @@ def main_loop(conf, foreground=False, task_manager_class=None):
             tm.log_error(traceback.get_traceback())
             tm.sleep()
 
+    if reexec:
```

Comment:
This looks like it'll work nicely in the typical case, but I'm wondering how do you want it to behave in the error case?

What I mean is: you send SIGHUP. Since it has to wait for the current worker's tasks to finish, it could still be some minutes or even hours before it goes into the ShutdownException case to break out of the above loop.  During that time, perhaps some other error occurs and it goes into the generic "except:" case and leaves the loop that way.

Do you still want it to re-exec in that case?

Reply:
I've reworked the PR to always reset the `reexec` flag whenever the worker process receives `SIGTERM`, `SIGINT` or has processed a `ShutdownWorker` task.

Reply:
Hmm, there is some weird bug that the worker gets stuck in a loop when a `ShutdownWorker` task gets assigned to it after the task manager was locked.

Reply:
Ok, I think I know where the problem is.  I'll post a separate PR with a fix.

Reply:
The issues I've encountered in https://github.com/release-engineering/kobo/pull/245#discussion_r1521786784, should be addressed in https://github.com/release-engineering/kobo/pull/251.